pkg/container/container.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package container
+
+import (
+	"github.com/google/go-containerregistry/pkg/authn"
+	containerregistry "github.com/google/go-containerregistry/pkg/crane"
+
+	"k8s.io/release/pkg/util"
+)
+
+const (
+	RegistryPassword = "REGISTRY_USER_PASSWORD"
+	RegistryUserName = "REGISTRY_USERNAME"
+)
+
+// Container is a wrapper around Container related functionality
+type Container struct {
+	client Client
+	Auth   containerAuth
+}
+
+type containerAuth struct {
```

Comment:
This extra level of indirection might be useful for testing or something (I haven't looked), but it seems like you could just have `Container.Auth` be an `authn.Basic` (though I also understand not wanting to mix structs from your dependencies into your exposed structs).